### PR TITLE
Improvements to data query section in README

### PR DESCRIPTION
**Pull request by AbhishekBhattacharya**
Changes made in accordance with the issue faced regarding **onSuccess** and **onError** that are functions that should be defined.Thus, should be added in the readme.

### CODE REVIEW

**Comment by coco98** [comment]
@AbhishekBhattacharya Thanks for the PR :) 
There are some mistakes in your README syntax. Some of the code is missing and ``` are showing up. I'm adding comments to your code commits to help you locate the mistakes.

You should always test/view your changes before you submit contributions! :)

File: README.md [hunk 1]
```diff
@@ -67,6 +67,46 @@ hasura.user // will be reset to anonymous user
 ### Data query
 
 **NOTE**: In the examples below, `onSuccess` and `onError` are callback functions that you must implement.
+    
+For example:
+    
+```
```
Comment (coco98) [inline README.md]: This should be:
```
```javascript
```
That will enable javascript syntax highlighting.

File: README.md [hunk 2]
```diff
@@ -67,6 +67,46 @@ hasura.user // will be reset to anonymous user
...
+Option 1:
+
+Use lambdas or anonymous functions directly for handling success/error.
+
```
Comment (coco98) [inline README.md]: Put the 2 lines above, outside the beginning ``` javascript
In the final README, this is how it should appear:

**Option 1:**
Use lambdas or anonymous functions directly for handling success/error.

```javascript
hasura.data.query
...
```

File: README.md [hunk 3]
```diff
@@ -67,6 +67,46 @@ hasura.user // will be reset to anonymous user
...
+hasura.data.query({
+  type: 'select',
+  args: {
+    table: 'article',
+    columns: ['*']
+  }},
+  (data) => { console.log(data); },
+  (error) => { console.log(error); }
+);
+
```
Comment (coco98) [inline README.md]: End the code fragment here by adding a  ```.

File: README.md [hunk 4]
```diff
@@ -67,6 +67,46 @@ hasura.user // will be reset to anonymous user
...
+Option 2:
+
+Use predefined functions as shown below:
+
+function mySuccessHandler (data) {
```
Comment (coco98) [inline README.md]: Begin a new code fragment above this line. Again, use ```javascript to start the code block.
Read more about code blocks here: https://help.github.com/articles/creating-and-highlighting-code-blocks/

**Review by coco98 — CHANGES_REQUESTED** [review]
I've added a lot of suggestions. Please note, that these are extremely simple changes that I could have just copy-pasted, but I hope this is a good learning experience for you!

**Comment by AbhishekBhattacharya** [comment]
Thanks, @coco98  for your help! It was a great experience :100: 
I have done your requested changes.Please review them and merge :+1:

File: README.md [hunk 5]
```diff
@@ -67,7 +67,49 @@ hasura.user // will be reset to anonymous user
 ### Data query
 
 **NOTE**: In the examples below, `onSuccess` and `onError` are callback functions that you must implement.
```
Comment (coco98) [inline README.md]: This line is redundant now right? You should remove it.

File: README.md [hunk 6]
```diff
@@ -67,7 +67,49 @@ hasura.user // will be reset to anonymous user
...
+  args: {
+    table: 'article',
+    columns: ['*']
+  }},
+  mySuccessHandler,
+  myErrorHandler
+);
+```
+
+The Data Query can be implemented as:
```
Comment (coco98) [inline README.md]: This portion is also redundant now. "The data query can be implemented as" and the entire code block below it too.

**Comment by coco98** [comment]
@AbhishekBhattacharya Added 2 more comments. Fix those and we should be good to merge!

**Review by coco98 — APPROVED** [review]

**Review by AbhishekBhattacharya — COMMENTED** [review]
Changed the Readme about handling onSuccess and onError callback functions

**Comment by AbhishekBhattacharya** [comment]
@coco98  Done! and thanks a lot for your guidance :)

**Pull request merged**